index.js

```diff
@@ -111,5 +111,10 @@ module.exports = {
 		}],
 		// require object literal shorthand syntax
 		'object-shorthand': ['error', 'always'],
+		// Warn when file extensions are not used in import paths
+		'import/extensions': ['warn', 'ignorePackages', {
+			// Require file extension for json imports
+			json: 'always',
```

Comment:
Redundant, no?

Reply:
99% of the time you don't import json from a package but in the rare case that you do as I have seen before you would not want
```js
import { joseph } from 'package/users'
```
but rather
```js
import { joseph } from 'package/users.json'
```

Reply:
99% of the time you don't tend to import json from a package but in the rare case that you do you probably wouldn't want
```js
import { joseph } from 'package/users'
```
but rather
```js
import { joseph } from 'package/users.json'
```

Reply:
99% of the time you don't tend to import json from a package but in the rare case that you do you probably wouldn't want
```js
import { joseph } from 'package/users'
```
but rather
```js
import { joseph } from 'package/users.json'
```

Reply:
99% of the time you don't tend to import json from a package but in the rare case that you do you probably wouldn't want
```js
import { joseph } from 'package/users'
```
but rather
```js
import { joseph } from 'package/users.json'
```
for clarity

Reply:
Yes, but the `ignorePackages` option already enforce it for json, no?

Reply:
> "ignorePackages"... rule enforces the use of extensions for all import statements except package imports

^from https://github.com/import-js/eslint-plugin-import/blob/v2.25.4/docs/rules/extensions.md#rule-details

Would mean `ignorePackages` would not enforce the .json extension if it's from a package is my understanding

Reply:
> "ignorePackages"... rule enforces the use of extensions for all import statements except package imports

^from https://github.com/import-js/eslint-plugin-import/blob/v2.25.4/docs/rules/extensions.md#rule-details

Would mean `ignorePackages` would not enforce the .json extension if it's from a package and would only enforce extensions for local imports is my understanding

Reply:
Hmm, that's not what I am understanding. For me package imports are:

```js
import package from 'package'
```

And the following would not be concerned with the `ignorePackages` rule, so `'package/users.json'` would be ok:

```js
import { joseph } from 'package/users.json'
```

Reply:
From the doc:


The following patterns are not considered problems when configuration set to `"ignorePackages"`:

```js
import express from 'express';
import foo from '@/foo'
```

The following patterns are not considered problems when configuration set to `['error', 'always', {ignorePackages: true} ]`:

```js
import baz from 'foo/baz.js'; // <-- this one in particular
import express from 'express';
import foo from '@/foo';
```


Maybe we need to go with the last option: `['warning', 'always', {ignorePackages: true} ]`:

Reply:
Agreed, latest push now uses that option

Reply:
Agreed, latest push now uses that option :)